src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -661,7 +677,8 @@ private static DateTime ConvertTime(DateTime dateTime, TimeZoneInfo sourceTimeZo
             // performance for the normal case at the expense of the 'ArgumentException'
             // case and Loss-less Local special cases.
             //
-            AdjustmentRule sourceRule = sourceTimeZone.GetAdjustmentRuleForTime(dateTime);
+            int? sourceRuleIndex;
+            AdjustmentRule sourceRule = sourceTimeZone.GetAdjustmentRuleForTime(dateTime, out sourceRuleIndex);
```

Comment:
Is there anything we can/should be validating here about the results, e.g.
`Debug.Assert(sourceRule == null || sourceRuleIndex.HasValue);`?